api/core/v2/tls.go

```diff
@@ -63,6 +70,10 @@ func (t *TLSOptions) ToTLSClientConfig() (*tls.Config, error) {
 		if err != nil {
 			return nil, err
 		}
+		// TODO(gbolo): we need an option to first load the system root CAs, then append the user specified CA(s).
+		// In this current form, our client will ONLY trust the specified CA(s).
+		// This may be desired by the user, but if this client config is used for a variety of endpoints,
+		// then perhaps the user should decide if they want system roots enabled or not.
```

Comment:
I think the rationale for not trying to combine the system's default CA certificate bundle and the user provided bundle is that the user can then completely override the system's default if they want or need to.

If users wants to combine the system's default bundle and a custom bundle, they can just create a new bundle containing the two? Or am I missing a pretty common use case that would really benefit from adding more flags? I'm in favor of simplicity and reducing the number of available flags! :)

Reply:
I agree. Using the operating system's certificate store should be the preferred approach, with the command line flag being a way to short-circuit that.

Reply:
Yea that makes sense. I will remove this comment then. Thanks guys